packages/amplication-gpt-gateway/src/conversationType/conversationType.service.ts

```diff
@@ -1,10 +1,41 @@
 import { Injectable } from "@nestjs/common";
+import { MessageInput } from "../dto/MessageInput";
+import { KafkaProducerService } from "../kafka/kafka.producer.service";
+import { MyMessageBrokerTopics } from "../kafka/topics";
 import { PrismaService } from "../prisma/prisma.service";
+import { TemplateService } from "../template/template.service";
 import { ConversationTypeServiceBase } from "./base/conversationType.service.base";
 
 @Injectable()
 export class ConversationTypeService extends ConversationTypeServiceBase {
-  constructor(protected readonly prisma: PrismaService) {
+  constructor(
+    protected readonly prisma: PrismaService,
+    protected readonly kafkaService: KafkaProducerService,
+    protected readonly templateService: TemplateService
+  ) {
     super(prisma);
   }
+
+  async startConversion(message: MessageInput): Promise<void> {
+    const { messageTypeId, params, requestUniqueId } = message;
+    const messageType = await this.prisma.conversationType.findUnique({
+      where: {
+        id: messageTypeId,
+      },
+    });
+
+    if (!messageType || !messageType.templateId)
+      throw new Error("Template not found");
+
+    const result = await this.templateService.processTemplateMessage({
+      templateId: messageType.templateId,
+      params,
+    });
+    if (!result) throw new Error("Template not found");
```

Comment:
I dont think we need to throw... instead we need to return the error to Amplication Server so the user will know that the process failed

Reply:
@yuval-hazaz  I looked at other implementations in our repo, and they all catch the error and write to the log. Does that sound like a good solution, or do you think that we should add it to the response ( ass status message)?